src/lib/Extension.ts

```diff
@@ -49,23 +51,57 @@ export default class Extension {
         targetDirectory: string,
         answers: Object,
     ) {
+        const tempDirectory = targetDirectory + '/temp';
         try {
             if (!fs.existsSync(targetDirectory)) {
                 createDirectory(targetDirectory);
+                createDirectory(tempDirectory);
             }
-            await execa('git', ['init'], { cwd: targetDirectory });
+            await execa('git', ['init'], { cwd: tempDirectory });
             await execa(
                 'git',
                 ['remote', 'add', 'origin', answers.project_url],
-                { cwd: targetDirectory },
+                { cwd: tempDirectory },
             );
-            await execa('git', ['pull', 'origin', 'main:main'], {
-                cwd: targetDirectory,
+            
+            await execa('git', ['pull', '--recurse-submodules', 'origin', 'main:main'], {
```

Comment:
Keep this branch name as configurable @manishprajapati52

Reply:
@vivek-gofynd 
Didn't understand this.
Configurable as in? 
Each project can have different branch?